asset/asset.go

```diff
@@ -425,6 +433,13 @@ func (g *GroupKey) IsEqual(otherGroupKey *GroupKey) bool {
 		g.Sig.IsEqual(&otherGroupKey.Sig)
 }
 
+// IsLocal returns true if the private key that corresponds to this group key
+// is held by this daemon. A non-local group key is stored with the tweaked
+// group key as the internal public key.
```

Comment:
This feels like mis-usage of those fields tbh. Also worth mentioning that I planned on removing the `GroupKey` struct as part of #91, since it is mostly the public key that is needed when not minting.

Reply:
The group key struct maps pretty directly to the `asset_groups` table structure in the DB, I think it's useful to simply usage there.

I agree this usage is awkward, could replace it with just a flag and then not require linking to an internal key?

Reply:
> Also worth mentioning that I planned on removing the GroupKey struct as part of https://github.com/lightninglabs/taro/issues/91 (simply replace it with a *btcec.PublicKey), since it is mostly the public key that is needed when not minting.

Orly? I think the struct is useful since at least for the minter, since it needs the plainkey and the tweak itself. Usually we refer to the tweaked key as the main group key, and that's sort of exposed on the RPC interface already. Curious to see what that would look like....particularly given that post #91 we'd sort of couple the group key and the sig more tightly (albeit in that same area). 

From another view, could make sense, since now the sig will only live at the genesis witness.

Reply:
I went a little back on this, since removing the struct could have some implications for usages other places (like in this PR). So for now just assume it will be kept :)

(went for documenting more clearly when the raw key and sig will be nil instead)